### PR TITLE
Go112

### PR DESCRIPTION
**Pull request by yangao22**
- Update Go1.12

- Update ruby and compass 

- Update git

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -22,14 +25,18 @@ RUN yum groupinstall -y -q "Development Tools"
...
 
 # Install the latest version of git
 RUN cd /tmp && \
-    wget https://github.com/git/git/archive/v2.7.0.tar.gz && \
-    tar xvfz ./v2.7.0.tar.gz && \
-    cd git-2.7.0 && \
-    make configure && \
+    wget https://mirrors.edge.kernel.org/pub/software/scm/git/git-2.9.5.tar.gz && \
+    tar xvfz ./git-2.9.5.tar.gz && \
+    cd git-2.9.5 && \
```
Comment (haggaret) [inline Dockerfile:36]: You don't need the make configure anymore?

File: Dockerfile [hunk 2]
```diff
@@ -1,6 +1,9 @@
 FROM signiant/docker-jenkins-centos-base:centos7-java8
 MAINTAINER devops@signiant.com
 
+EXPOSE 8000
+EXPOSE 9000
```
Comment (haggaret) [inline Dockerfile]: Why do these ports need to be exposed?

**Review by haggaret — APPROVED** [review]

File: Dockerfile [hunk 3]
```diff
@@ -1,6 +1,9 @@
 FROM signiant/docker-jenkins-centos-base:centos7-java8
 MAINTAINER devops@signiant.com
 
+EXPOSE 8000
```
Comment (miketypeguy) [inline Dockerfile]: Why do these ports have to be exposed?

File: Dockerfile (hunk 2 above)
Comment (miketypeguy) [reply Dockerfile]: Great minds

File: Dockerfile (hunk 1 above)
Comment (yangao22) [reply Dockerfile:36]: not anymore

File: Dockerfile [hunk 4]
```diff
@@ -51,8 +58,9 @@ ENV UMPIRE_VERSION 0.5.3
 # Install umpire
 RUN pip2.7 install umpire==${UMPIRE_VERSION}
 
-#Install go
-RUN yum install -y golang
+# Install golang 1.12
+RUN wget https://storage.googleapis.com/golang/go1.12.1.linux-amd64.tar.gz -O /tmp/go1.12.1.linux-amd64.tar.gz
```
Comment (shanebarnes) [inline Dockerfile]: 1.12.3 is the latest stable (https://dl.google.com/go/go1.12.3.linux-amd64.tar.gz)

File: Dockerfile (hunk 3 above)
Comment (yangao22) [reply Dockerfile]: I think Dave made the change in 2018-10-15: Only expose the actual needed ports of it breaks the docker jenkins plugin with an object too big exception. 
But I think I duplicated the ports too. Will remove it

**Review by shanebarnes — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (yangao22) [reply Dockerfile:36]: I remember that cause the error on centos6, I'll check if that works with `make config`

File: Dockerfile (hunk 2 above)
Comment (yangao22) [reply Dockerfile]: Dave made the change in 2018-10-15: Only expose the actual needed ports of it breaks the docker jenkins plugin with an object too big exception.
But I think I duplicated the ports too. Will remove it

**Review by miketypeguy — APPROVED** [review]

**Pull request merged**